lib/ansible/modules/notification/office_365_connector_card.py

```diff
@@ -0,0 +1,291 @@
+OFFICE_365_CARD_TYPE = "MessageCard"
+
+
+def build_actions(actions):
+    action_items = []
+
+    for action in actions:
+        action_item = snake_dict_to_camel_dict(action)
+        action_items.append(action_item)
+
+    return action_items
+
+
+def build_sections(sections):
+    sections_created = []
+
+    for section in sections:
+        sections_created.append(build_section(section))
+
+    return sections_created
+
+
+def build_section(section):
+    section_payload = dict()
+
+    if 'title' in section:
+        section_payload['title'] = section['title']
+
+    if 'start_group' in section:
+        section_payload['startGroup'] = section['start_group']
+
+    if 'activity_image' in section:
+        section_payload['activityImage'] = section['activity_image']
+
+    if 'activity_title' in section:
+        section_payload['activityTitle'] = section['activity_title']
+
+    if 'activity_subtitle' in section:
+        section_payload['activitySubtitle'] = section['activity_subtitle']
+
+    if 'activity_text' in section:
+        section_payload['activityText'] = section['activity_text']
+
+    if 'hero_image' in section:
+        section_payload['heroImage'] = section['hero_image']
+
+    if 'text' in section:
+        section_payload['text'] = section['text']
+
+    if 'facts' in section:
+        section_payload['facts'] = section['facts']
+
+    if 'images' in section:
+        section_payload['images'] = section['images']
+
+    if 'actions' in section:
+        section_payload['potentialAction'] = build_actions(section['actions'])
+
+    return section_payload
+
+
+def build_payload_for_connector_card(module, summary, color, title, text, actions, sections):
+    payload = dict()
+    payload['@context'] = OFFICE_365_CARD_CONTEXT
+    payload['@type'] = OFFICE_365_CARD_TYPE
+
+    if summary is not None:
+        payload['summary'] = summary
+
+    if color is not None:
+        payload['themeColor'] = color
+
+    if title is not None:
+        payload['title'] = title
+
+    if text is not None:
+        payload['text'] = text
+
+    if actions:
+        payload['potentialAction'] = build_actions(actions)
+
+    if sections:
+        payload['sections'] = build_sections(sections)
+
+    payload = module.jsonify(payload)
+    return payload
+
+
+def do_notify_connector_card_webhook(module, webhook, payload):
+    headers = {
+        'Content-Type': 'application/json'
+    }
+
+    response, info = fetch_url(
+        module=module,
+        url=webhook,
+        headers=headers,
+        method='POST',
+        data=payload
+    )
+
+    if info['status'] == 200:
+        module.exit_json(changed=True)
+    else:
+        module.fail_json(
+            msg="failed to send %s as a connector card to Incoming Webhook: %s"
+                % (payload, info['msg'])
+        )
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            webhook=dict(required=True, no_log=True),
```

Comment:
@abadger I wasn't aware. I wonder why some of the modules with no type seem to work as if they have type='raw', and fix those. Sorry for the noise.